### DISCUSSION TITLE
Execution failed for task ':graphqlIntrospectSchema'

### DISCUSSION THREAD

**Discussion by DmitryevichD** [ExpediaGroup/graphql-kotlin#980] [Q&A] [answered]
Hi!
I want to use your plugin in my project. I read the manual, but can't assemble the project. 
When I run ./gradlew build I have the problem: 
```
Execution failed for task ':graphqlIntrospectSchema'.
> kotlin/coroutines/AbstractCoroutineContextKey
```
[My Project](https://github.com/DmitryevichD/graphql-not-work)

Can you help me with the problem?

Thanks!

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
That error generally implies some classpath issue. My guess is that the issue is from Kotlin version mismatch -> `graphql-kotlin` plugin attempts to pull in version `1.3.x` whereas you apply Kotlin 1.4 plugin. Can you downgrade to Kotlin version `1.3.72` and see whether this fixes your issue?

**Reply by DmitryevichD** [reply]
I downgraded the version to 1.3.72, but this case didn't work too.
After that, I tried starting gradle-client from the repository. 
I started the server and the client, and
```
Execution failed for task ':graphqlGenerateClient'.
> unable to find corresponding field definition of retrieveBasicObject in Query
```
Please help me run your examples

**Reply by dariuszkuc** [reply]
The above error implies that your queries do not match the target server (looks like you are using queries from [example gradle client](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples/client/gradle-client) which should work against the [sample server](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples/client/server)). Make sure that your queries are written against a valid server.

**Reply by DmitryevichD** [reply]
Thank you for your answer, can I ask one more? :)
I should use Rest API and GraphQL together. Can I add some property, for example as graphql-java

```
graphql:
  servlet:
    mapping: /my/custom/graphql/mapping
```
for the dispatcher servlet to make a mapping request to graphql?

P.S.

```
graphql:
   endpoint: /my/custom/endpoint
```
this case doesn't work, dispatcher servlet doesn't know how to mapping a path

**Reply by dariuszkuc** [reply]
Unfortunately you won't be able to use `graphql-kotlin-spring-server` with a servlet libraries. We've built our library on top of Spring Webflux (reactive) which is an alternative to Spring MVC (blocking/servlet) and you cannot use both in the same project - see https://expediagroup.github.io/graphql-kotlin/docs/spring-server/spring-overview#webflux-vs-webmvc.

You definitely can expose your REST endpoints using Spring Webflux (e.g. by default `graphql-kotlin-spring-server` exposes `/sdl`). See [Spring docs for details](https://docs.spring.io/spring-framework/docs/current/reference/html/web-reactive.html).
